diff --git a/recipes-core/aziotd/files/aziot-keyd.service b/recipes-core/aziotd/files/aziot-keyd.service
index fa2af87..c3bcb9b 100644
--- a/recipes-core/aziotd/files/aziot-keyd.service
+++ b/recipes-core/aziotd/files/aziot-keyd.service
@@ -5,7 +5,7 @@ After=aziot-keyd.socket
 
 [Service]
 Environment="AZIOT_LOG=error"
-Environment="LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/rust"
+Environment="LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/rustlib/aarch64-tdx-linux-gnu/lib"
 ExecStart=/usr/bin/aziot-keyd
 KillMode=process
 Restart=on-failure
