diff --git a/recipes-core/aziotd/files/aziot-tpmd.service b/recipes-core/aziotd/files/aziot-tpmd.service
index a983636..68c09a9 100644
--- a/recipes-core/aziotd/files/aziot-tpmd.service
+++ b/recipes-core/aziotd/files/aziot-tpmd.service
@@ -5,7 +5,7 @@ After=aziot-tpmd.socket
 
 [Service]
 Environment="AZIOT_LOG=error"
-Environment="LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/rust"
+Environment="LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/rustlib/aarch64-tdx-linux-gnu/lib"
 ExecStart=/usr/bin/aziot-tpmd
 KillMode=process
 Restart=on-failure
